dashboard/app/models/user.rb

```diff
@@ -474,7 +474,7 @@ def self.find_or_create_facilitator(params, invited_by_user)
 
   has_many :section_instructors, foreign_key: 'instructor_id'
   has_many :active_section_instructors, -> {where(status: :active)}, class_name: 'SectionInstructor', foreign_key: 'instructor_id'
-  has_many :sections_instructed, through: :active_section_instructors, source: :section
+  has_many :sections_instructed, -> {without_deleted}, through: :active_section_instructors, source: :section
```

Comment:
Are we sure we want this on just the `sections_instructed` and not `section_instructors` or `active_section_instructors`? Shouldn't section_instructors be deleted when a section is deleted?

Reply:
Good questions!

1) `SectionInstructors` are (soft) deleted when a section is deleted. The `dependent: :destroy` in (section.rb)[https://github.com/code-dot-org/code-dot-org/blob/c33fd3912d18cba4e8730b88713dd421fbe998ff/dashboard/app/models/sections/section.rb#L67] accomplishes this. I just verified this behavior in the console.
2) The `active_section_instructors` association respects the soft-deletion on its own, it's only when combined in a `has_many: :through` association that this doesn't work, so that's where I'm targeting the fix.
<img width="961" alt="image" src="https://github.com/code-dot-org/code-dot-org/assets/4108328/8e5b0e3c-21a4-4b77-b168-0610b0126d7c">

Reply:
Very interesting